packages/react/babel.ts

```diff
@@ -2,12 +2,44 @@
  * Babel preset to provide React support for Nx.
  */
 
-module.exports = function (api: any, options: {}) {
+interface ReactBabelOptions {
+  runtime?: string;
+  importSource?: string;
+}
+
+module.exports = function (api: any, options: ReactBabelOptions) {
   api.assertVersion(7);
+  const env = api.env();
+
   return {
     presets: [
       '@nrwl/web/babel',
-      [require.resolve('@babel/preset-react'), { useBuiltIns: true }],
+      [
+        require.resolve('@babel/preset-react'),
+        getReactPresetOptions({
+          presetOptions: options,
+          env,
+        }),
+      ],
     ],
   };
 };
+
+function getReactPresetOptions({ presetOptions, env }) {
+  const reactPresetOptions: Record<string, string | boolean> = {
+    runtime: presetOptions.runtime ?? 'automatic',
+    development: env !== 'production',
+  };
+
+  // JSX spread is transformed into object spread in `@babel/plugin-transform-react-jsx`
+  // `useBuiltIns` will be removed in Babel 8.
+  if (presetOptions.runtime === 'automatic') {
```

Comment:
since `presetOptions.runtime` is nullable, but `reactPresetOptions.runtime` is always set, should this be checking `reactPresetOptions.runtime` instead?

Reply:
That's a good point, especially since we're not migrating their `.babelrc` to set `runtime` so we need to support it being undefined.